Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -220,7 +234,20 @@ public struct HTTPHeaders: CustomStringConvertible {
     // Because we use CoW implementations HTTPHeaders is also CoW
     private var buffer: ByteBuffer
     private var headers: [HTTPHeader]
-    private var continuous: Bool = true
+
+    fileprivate var state: HTTPHeadersModificationState = .notModified
+
+    /// Update the current state of the headers if needed.
+    private mutating func updateStateIfNeeded(newState: HTTPHeadersModificationState) {
+        switch newState {
+        case .modified where self.state == .notModified, .modifiedNonContinuous:
+            self.state = newState
+        case .notModified:
+            fatalError("Should never be called with \(newState)")
+        default:
+            break
+        }
```

Comment:
I tend to think this kind of method is nicer on the enum. Also no need to say "ifNeeded" in the name.

Reply:
In fact, it may be nicer just to have functions that explain what they are, e.g. `state.appendedHeader()` and `state.removedHeader()`. Then you can have some computed vars that use the right names, e.g `state.continuous` and `state.wasModified`.